cap-ci/pipeline.template

```diff
@@ -67,6 +69,46 @@ resources:
 
 {{ $pipelineName := .Env.PIPELINE }}
 
+{{- $create_k8s := `
+            # create k8s cluster
+            export random_variable=$(hexdump -n 8 -e '2/4 "%08x"' /dev/urandom)
+            if [[ ${BACKEND} == "gke" ]]; then
+              printf "%s" '((gke-key-json))' > $PWD/gke-key.json
+              export GKE_CRED_JSON=$PWD/gke-key.json
+              export GKE_CLUSTER_NAME=ci-${BACKEND}-${random_variable}
+              export CLUSTER_NAME=${GKE_CLUSTER_NAME}
+              (cd catapult; make BACKEND=$BACKEND GKE_CREDS_JSON=$GKE_CRED_JSON GKE_CLUSTER_NAME=${GKE_CLUSTER_NAME} CLUSTER_NAME=${GKE_CLUSTER_NAME})
+              echo ${GKE_CLUSTER_NAME} > kubeconfig-pool/name
+              cp catapult/build${GKE_CLUSTER_NAME}/kubeconfig kubeconfig-pool/metadata
```

Comment:
> I would only copy the strictly necessary. For example,
> 
> * `buildfolder/kubeconfig`
> * `buildfolder/cap-terraform/gke/`
> 

What do you mean? I am only copying kubeconfig file which makes the pool

> I would not copy and check into cf-ci-pools.git:
> 
> * `buildfolder/bin/*`
> * nor the terraform binary plugins that get downloaded with `terraform init`. Example in the case of caasp:
> 

where do you see these getting copied? 

> ```
> 
> 
> vic@viccuad ~/suse/catapult-pools/buildcaasp4-pool1-cap-ci/deployment/.terraform$ tree -h
> .
> └── [4.0K]  plugins
>     └── [4.0K]  linux_amd64
>         ├── [ 245]  lock.json
>         ├── [ 19M]  terraform-provider-null_v2.1.2_x4
>         ├── [ 31M]  terraform-provider-openstack_v1.27.0_x4
>         └── [ 20M]  terraform-provider-template_v2.1.2_x4
> ```
> 
> Also, if we are commiting things to cf-ci-pools.git, we should checkout the pool resource with `params: {depth: 10}` or similar.

Reply:
This is one approach that could be followed to correctly delete clusters with catapult, and not use the CI to call `gcloud delete` directly.

Reply:
Haven't tested this yet, but does the pool resource actually allow you to use a directory with files in it as the lockfile? Because when you `get` a pool resource the name and 'data' go into `name` and `metadata` files in the concourse task.. just wanted to check if you've tested this yet

Reply:
The pool resource doesn't seem to accept folders, yes.

No, I haven't tested it. The work for this PR or the card jira.suse.com/browse/CAP-1279 seems the moment were we need to implement it.

I'm getting weird ideas:
- tar the `buildfolder` and add that to the pool
- create a different pool resource that accepts folders
- insert links/tokens into the pool, and save the terraform files elsewhere

I really think we need a way to save the terraform files and call `terraform destroy`. On GKE we are lucky we can delete the cluster manually, but I don't think this applies to AWS, AKS, or Openstack.

Reply:
@viccuad yes I was thinking about tarring the buildir also, might be a solution we have to go with eventually (also I don't understand why it was so hard to find this thread in github's UI)

Reply:
I think tarring would be preferable to creating a custom resource, not that there's anything wrong with creating a custom resource; I think we should just prefer working with the standard ones when possible

Reply:
I agree about working with standard resources.